ctapipe/image/ellipsoid.py

```diff
@@ -0,0 +1,443 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: UTF-8 -*-
+"""
+Shower image parametrization based on image fitting.
+"""
+
+from enum import Enum
+
+import astropy.units as u
+import numpy as np
+from astropy.coordinates import Angle
+from iminuit import Minuit
+
+from ctapipe.image.cleaning import dilate
+from ctapipe.image.hillas import hillas_parameters
+from ctapipe.image.pixel_likelihood import (
+    mean_poisson_likelihood_gaussian,
+    neg_log_likelihood_approx,
+)
+from ctapipe.image.toymodel import SkewedGaussian, SkewedGaussianLaplace
+
+from ..containers import ImageFitParametersContainer
+
+PED_TABLE = {
+    "LSTCam": 2.8,
+    "NectarCam": 2.3,
+    "FlashCam": 2.3,
+    "SST-Camera": 0.5,
+    "CHEC": 0.5,
+    "DUMMY": 0,
+    "testcam": 0,
+}
+SPE_WIDTH = 0.5
+
+__all__ = [
+    "image_fit_parameters",
+    "ImageFitParameterizationError",
+    "PDFType",
+]
+
+
+class PDFType(Enum):
+    gaussian = "gaussian"
+    laplace = "laplace"
+    skewed = "skewed"
+
+
+def create_initial_guess(geometry, image, size):
+    """
+    This function computes the seeds of the fit with the Hillas parameters
+    Parameters
+    ----------
+    geometry: ctapipe.instrument.CameraGeometry
+        Camera geometry, the cleaning mask should be applied to improve performance
+    image: ndarray
+        Charge in each pixel, the cleaning mask should already be applied to
+        improve performance.
+    size: float/int
+        Total charge after cleaning and dilation
+    Returns
+    -------
+    initial_guess: dict
+    Seed parameters of the fit.
+    """
+    unit = geometry.pix_x.unit
+    hillas = hillas_parameters(geometry, image)  # compute Hillas parameters
+
+    initial_guess = {}
+
+    if unit.is_equivalent(u.m):
+        initial_guess["cog_x"] = hillas.x.to_value(unit)
+        initial_guess["cog_y"] = hillas.y.to_value(unit)
+    else:
+        initial_guess["x"] = hillas.fov_lon.to_value(unit)
+        initial_guess["y"] = hillas.fov_lat.to_value(unit)
+
+    initial_guess["length"] = hillas.length.to_value(unit)
+    initial_guess["width"] = hillas.width.to_value(unit)
+    initial_guess["psi"] = hillas.psi.value
+
+    initial_guess["skewness"] = hillas.skewness
+
+    if (hillas.width.to_value(unit) == 0) or (hillas.length.to_value(unit) == 0):
+        raise ImageFitParameterizationError("Hillas width and/or length is zero")
+
+    initial_guess["amplitude"] = size
+    return initial_guess
+
+
+def dilation(n, cleaned_mask, geometry):
```

Comment:
Pretty sure this already exists in `ctapipe.image.cleaning`, no need to redefine

Reply:
actually, we only have `dilate` which does a single dilation. THe loop for multiple is not there, but this code should in any case go into `cleaning.py` (maybe call it `dilate_multiple`)

Reply:
or use `functools.reduce` to do the dilate(dilate(dilate()) call chaining